src/model/model_views.nit

```diff
@@ -0,0 +1,234 @@
+			mclasses.add mclass
+		end
+		return mclasses
+	end
+
+	# MClassDefs visible through `self`.
+	var mclassdefs: Set[MClassDef] is lazy do
+		var mclassdefs = new HashSet[MClassDef]
+		for mclass in mclasses do
+			for mclassdef in mclass.mclassdefs do
+				if not accept_mentity(mclassdef) then continue
+				mclassdefs.add mclassdef
+			end
+		end
+		return mclassdefs
+	end
+
+	# MProperties visible through `self`.
+	var mproperties: Set[MProperty] is lazy do
+		var mproperties = new HashSet[MProperty]
+		for mproperty in model.mproperties do
+			if not accept_mentity(mproperty) then continue
+			mproperties.add mproperty
+		end
+		return mproperties
+	end
+
+	# MPropdefs visible through `self`.
+	var mpropdefs: Set[MPropDef] is lazy do
+		var mpropdefs = new HashSet[MPropDef]
+		for mproperty in mproperties do
+			for mpropdef in mproperty.mpropdefs do
+				if not accept_mentity(mpropdef) then continue
+				mpropdefs.add mpropdef
+			end
+		end
+		return mpropdefs
+	end
+
+	# Lists all MEntities visible through `self`.
+	var mentities: Set[MEntity] is lazy do
+		var res = new HashSet[MEntity]
+		res.add_all mpackages
+		res.add_all mgroups
+		res.add_all mmodules
+		res.add_all mclasses
+		res.add_all mclassdefs
+		res.add_all mproperties
+		res.add_all mpropdefs
+		return res
+	end
+
+	private fun init_visitor(v: ModelVisitor) do
+		v.min_visibility = self.min_visibility
+		v.include_fictive = self.include_fictive
+		v.include_empty_doc = self.include_empty_doc
+		v.include_attribute = self.include_attribute
+		v.include_test_suite = self.include_test_suite
+	end
+
+	# Searches MEntities that match `name`.
+	fun mentities_by_name(name: String): Array[MEntity] do
+		var res = new Array[MEntity]
+		for mentity in mentities do if mentity.name == name then res.add mentity
+		return res
+	end
+
+	# Looks up a MEntity by its full `namespace`.
+	#
+	# Usefull when `mentities_by_name` returns conflicts.
+	#
+	# Namespaces must be of the form `package::core::module::Class::prop`.
+	fun mentities_by_namespace(namespace: String): Array[MEntity] do
+		var v = new LookupNamespaceVisitor(namespace)
+		init_visitor(v)
+		for mpackage in mpackages do
+			v.enter_visit(mpackage)
+		end
+		return v.results
+	end
+
+	# Build an concerns tree with from `self`
+	fun to_tree: MEntityTree do
+		var v = new ModelTreeVisitor
+		init_visitor(v)
+		for mpackage in mpackages do
+			v.enter_visit(mpackage)
+		end
+		return v.tree
+	end
+end
+
+class LookupNamespaceVisitor
+	super ModelVisitor
+
+	var namespace: String
+
+	private var parts: Array[String] is lazy do return namespace.split_with("::")
+
+	var results = new Array[MEntity]
+
+	redef fun visit(mentity) do mentity.accept_namespace_visitor(self)
+end
+
+class ModelTreeVisitor
+	super ModelVisitor
+
+	var tree = new MEntityTree
+
+	redef fun visit(mentity) do mentity.accept_tree_visitor(self)
+end
+
+redef class MEntity
+
+	# Get a public view of the model
+	fun public_view: ModelView do
```

Comment:
Because `Model` isa `MEntity` so it will be available for everyone that has a Model reference.

Also most tools (or at least the methods coded in these tools) does not always have a reference to a `Model` instance but only a `MModule` or something else.
So if the method is only in `Model` one as to write `mmodule.model.public_view` instead of `mmodule.public_view`.

This is why.